### PR TITLE
Refactor Logic into kindle_highlights.py

### PR DESCRIPTION
**Pull request by kelvindo**
- Separate out parsing and sampling logic into kindle_highlights.py to make it easier to test
 - Add a test file test_kindle_highlights.py to make it easier to locally test sampling and see result
 - Change all references of "quotes" to "highlights" for consistency

### CODE REVIEW

File: kindle_highlights.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import random
+
+MIN_HIGHLIGHTS = 5
+NUM_BOOKS_TO_SAMPLE = 3
+NUM_HIGHLIGHTS_TO_SAMPLE = 2
+
+def parse_highlights(highlights_json):
+    books_to_highlights = {}
+    for book in highlights_json["books"]:
+        # Ignore books that have fewer than MIN_HIGHTLIGHTS highlights
```
Comment (japalio) [inline kindle_highlights.py:12]: Do we always want to ignore in this case?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*.pyc
```
Comment (japalio) [inline .gitignore:1]: What does this do?

File: kindle_highlights.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    weight_list = [math.sqrt(len(highlights[book])) for book in book_list]
+    selected_highlights = {}
+    books = weighted_sample(book_list, weight_list, NUM_BOOKS_TO_SAMPLE)
+    for book in books:
+        sampled_highlights = random.sample(highlights[book], NUM_HIGHLIGHTS_TO_SAMPLE)
+        selected_highlights[book] = sampled_highlights
+
+    return selected_highlights
+
+def weighted_sample(population, weights, num):
```
Comment (japalio) [inline kindle_highlights.py]: Can you add comments explaining how the weighting works?

File: .gitignore (hunk 2 above)
Comment (kelvindo) [reply .gitignore:1]: It's the gitignore

File: kindle_highlights.py (hunk 1 above)
Comment (kelvindo) [reply kindle_highlights.py:12]: Yes

File: kindle_highlights.py (hunk 3 above)
Comment (kelvindo) [reply kindle_highlights.py]: No

Comment (kelvindo) [reply kindle_highlights.py]: Fine.

File: kindle_highlights.py (hunk 1 above)
Comment (kelvindo) [reply kindle_highlights.py:12]: We can change it but I find that that books with fewer than 5 highlights tend to be books I just read the sample. Usually its the same quote every time too making these random selections less useful.

**Review by japalio — APPROVED** [review]

**Pull request merged**